lib/inputs/number-input.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+var _ = require('underscore');
+var BaseInput = require('./base-input');
+var errors = require('../errors');
+
+const DEFAULT_VALUE = 0;
+
+class NumberInput extends BaseInput {
+    constructor(id, params) {
+        super('number', id, params);
```

Comment:
Instead of having a different constructor signature between the `BaseInput` and subclasses, I'd have the subclasses implement a

```
getType() {
  return 'number';
}
```

or explicitly set `this.type` in the subclass constructor.